packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-endpoint-config.ts

```diff
@@ -0,0 +1,135 @@
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+  ];
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: SageMakerCreateEndpointConfigProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern || sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, SageMakerCreateEndpointConfig.SUPPORTED_INTEGRATION_PATTERNS);
+
+    // verify production variants
+    if (props.productionVariants.length < 1 || props.productionVariants.length > 10) {
+      throw new Error('Must specify from 1 to 10 production variants per endpoint configuration');
+    }
+    props.productionVariants.forEach((variant) => {
+      if ( variant.initialInstanceCount < 1) throw new Error('Must define at least one instance');
+      if ( variant.initialVariantWeight && variant.initialVariantWeight <= 0) {
+        throw new Error('InitialVariantWeight has minimum value of 0');
+      }
+    });
+    this.taskPolicies = this.makePolicyStatements();
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('sagemaker', 'createEndpointConfig', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject(this.renderParameters()),
+    };
+  }
+
+  private renderParameters(): { [key: string]: any } {
+    return {
+      EndpointConfigName: this.props.endpointConfigName,
+      ...this.props.kmsKeyId ? { KmsKeyId: this.props.kmsKeyId } : {},
+      ...this.renderProductionVariants(this.props.productionVariants),
+      ...this.renderTags(this.props.tags),
+    };
+  }
+
+  private makePolicyStatements(): iam.PolicyStatement[] {
+    const stack = cdk.Stack.of(this);
+    // https://docs.aws.amazon.com/sagemaker/latest/dg/api-permissions-reference.html
+    return [
+      new iam.PolicyStatement({
+        actions: ['sagemaker:CreateEndpointConfig'],
+        resources: [
+          stack.formatArn({
+            service: 'sagemaker',
+            resource: 'endpoint-config',
+            // If the endpoint configuration name comes from input, we cannot target the policy to a particular ARN prefix reliably.
+            // SageMaker uses lowercase for resource name in the arn
+            resourceName: sfn.JsonPath.isEncodedJsonPath(this.props.endpointConfigName) ? '*' : `${this.props.endpointConfigName.toLowerCase()}`,
+          }),
+        ],
+      }),
+      new iam.PolicyStatement({
+        actions: ['sagemaker:ListTags'],
+        resources: [
+          stack.formatArn({
+            service: 'sagemaker',
+            resource: 'endpoint-config',
+            resourceName: '*',
```

Comment:
can this be scoped down any further?
if resource level scoping is not possible we should link to any relevant documentation and ensure it's captured somewhere

Reply:
actually i think I scoped this permission too much. Based on https://docs.aws.amazon.com/sagemaker/latest/dg/api-permissions-reference.html for `sagemaker:ListTags` the resource is `arn:aws:sagemaker:region:account-id:*`, also this permission is similar to the one defined on for the training job in https://docs.aws.amazon.com/step-functions/latest/dg/sagemaker-iam.html#sagemaker-iam-createtrainingjob :
`    {
      "Effect": "Allow",
      "Action": [
        "sagemaker:ListTags"
      ],
      "Resource": [
        "*"
      ]
    },`